sky/backends/backend_utils.py

```diff
@@ -1803,38 +1799,41 @@ def _update_cluster_status_no_lock(
         return record
 
     cluster_name = handle.cluster_name
-    try:
-        # TODO(zhwu): This function cannot distinguish transient network error
-        # in ray's get IPs vs. ray runtime failing.
-        external_ips = handle.external_ips(use_cached_ips=False)
-        # This happens to a stopped TPU VM as we use gcloud to query the IP.
-        if external_ips is None or len(external_ips) == 0:
-            raise exceptions.FetchIPError(
-                reason=exceptions.FetchIPError.Reason.HEAD)
-        if handle.launched_nodes == 1:
-            # Check the ray cluster status. We have to check it for single node
-            # case, since the get_node_ips() does not require ray cluster to be
-            # running.
-            ssh_credentials = ssh_credential_from_yaml(handle.cluster_yaml)
-            runner = command_runner.SSHCommandRunner(external_ips[0],
-                                                     **ssh_credentials)
-            returncode = runner.run('ray status', stream_logs=False)
-            if returncode:
+    use_spot = handle.launched_resources.use_spot
+    # if cluster is not spot, we can determine its health by "ray status".
```

Comment:
This change makes sense to me. Cc @Michaelvll to take another look.

Can we expand the comment here? E.g., mention including the following
- For non-spot clusters, this is an optimization: we call external_ips() and/or SSH into the cluster to run 'ray status' in various cases to determine cluster health, because these may be faster than querying the true node statuses from the cloud provider.
- For spot clusters, the above can be unsafe. <Fill in the detailed GCP scenario that this PR fixes.> Therefore we directly query the cloud provider.

Reply:
Good point. I've updated the comments.